File: clients/ruby/model/city_cell.rb
module Model

# City cell
module CityCell
    # Road
    ROAD = 0
    # Building
    BUILDING = 1
    # Refill station
    REFILL_STATION = 2

    # Read CityCell from input stream
    def self.read_from(stream)
        result = stream.read_int()
        if result < 0 || result >= 3
            raise "Unexpected tag value"
        end
        result
    end

    def self.to_s(value)
        if value == ROAD
            return "ROAD"
        end
        if value == BUILDING
            return "BUILDING"
        end
        if value == REFILL_STATION
            return "REFILL_STATION"
        end
        raise "Impossible happened"
    end

    def self.to_str(value)
        self.to_s(value)
    end
end

end

File: clients/php/Model/CityCell.php
<?php

namespace Model {
    require_once 'Stream.php';

    /**
     * City cell
     */
    abstract class CityCell
    {
        /**
         * Road
         */
        const ROAD = 0;

        /**
         * Building
         */
        const BUILDING = 1;

        /**
         * Refill station
         */
        const REFILL_STATION = 2;

        /**
         * Read CityCell from input stream
         */
        public static function readFrom(\InputStream $stream): int
        {
            $result = $stream->readInt32();
            if (0 <= $result && $result < 3) {
                return $result;
            }
            throw new Exception('Unexpected tag value');
        }
    }
}